client/src/client.js

```diff
@@ -151,139 +177,100 @@ class Client {
-
-    // Choose one voucher - rules to select between vouchers: use cheapest priced and then with the one that has the earliest endDate
-    const apiVoucher = apiVouchers.sort((a, b) => {
-      const rightA = this.orejs.getRight(a, apiName);
-      const rightB = this.orejs.getRight(b, apiName);
-      return rightA.price_in_cpu - rightB.price_in_cpu || a.instrument.start_time - b.instrument.end_time
-    })[apiVouchers.length - 1];
-
-    const apiRight = this.orejs.getRight(apiVoucher, apiName);
-
-    return { apiVoucher, apiRight };
-  }
+    Object.keys(urlParameters).forEach(key => {
+      url.searchParams.append(key, urlParameters[key])
+    });
 
-  //Call the Verifier to approve the request
-  async getAccessTokenFromVerifier(apiVoucher, apiRight, encryptedParams) {
-    let errorMessage;
-    let result;
-    const signature = await this.orejs.signVoucher(apiVoucher.id);
-
-    const options = {
-      method: 'POST',
-      body: JSON.stringify({
-        requestParams: encryptedParams,
-        rightName: apiRight.right_name,
-        signature,
-        voucherId: apiVoucher.id
-      }),
+    options = {
+      method: httpMethod,
+      body: JSON.stringify(bodyParameters),
       headers: {
-        'Content-Type': 'application/json'
+        'Content-Type': 'application/json',
+        'Ore-Access-Token': oreAccessToken
       }
     };
 
-    // Call the Verifier to approve the request
-    try {
-      result = await fetch(`${this.config.verifier}/verify`, options);
-      if (!result.ok) {
-        let error = await result.json();
-        throw new Error(error.message);
-      }
-    } catch (error) {
-      errorMessage = "Internal Server Error";
-      throw new Error(`${errorMessage}:${error.message}`);
-    }
+    return {
+      url,
+      options
+    };
 
-    const { endpoint, oreAccessToken, method, additionalParameters, accessTokenTimeout } = await result.json();
+  }
 
-    if (!oreAccessToken || oreAccessToken === undefined) {
-      errorMessage = "Internal Server Error: Verifier is unable to return an ORE access token. Make sure a valid voucher is passed to the verifier."
-      throw new Error(`${errorMessage}`);
-    }
+  async getInstrumentAndRight(rightName) {
+    // Call orejs.findInstruments(accountName, activeOnly:true, category:’apiMarket.apiVoucher’, rightName:’xxxx’) => [instruments]
+    const instruments = await this.orejs.findInstruments(this.config.accountName, true, this.config.instrumentCategory, rightName)
 
-    if (!endpoint || endpoint === undefined) {
-      errorMessage = "Internal Server Error: Verifier is unable to find the Api endpoint. Make sure to pass in the correct right name you want to access."
-      throw new Error(`${errorMessage}`);
-    }
+    // Choose one voucher - rules to select between vouchers: use cheapest priced and then with the one that has the earliest endDate
+    const instrument = instruments.sort((a, b) => {
```

Comment:
I would think this rule should be in OREJS with an option to specify that that caller wants the lowest cost option